wolfcrypt/test/test.c

```diff
@@ -25012,10 +25012,73 @@ static int crypto_ecc_sign(const uint8_t *key, uint32_t keySz,
 
     return ret;
 }
+#endif /* HAVE_ECC_SIGN && HAVE_ECC_VERIFY */
+
+#ifdef HAVE_ECC_DHE
+/* 
+ * Currently we only have support for P-256/384 for non-blocking key gen and
+ * shared secret gen.
+ */
+static int ecc_test_nonblock_dhe(WC_RNG* rng) {
+    int ret = 0;
+    ecc_key keyA;
+    ecc_key keyB;
+    ecc_nb_ctx_t nbCtxA;
+    ecc_nb_ctx_t nbCtxB;
+    byte secretA[ECC_SHARED_SIZE];
+    byte secretB[ECC_SHARED_SIZE];
+    word32 secretSzA = ECC_SHARED_SIZE;
+    word32 secretSzB = ECC_SHARED_SIZE;
+
+    if (ret == 0) {
```

Comment:
This is "always true". Some static analyzers will complain. Please remove this check for the first command. Also you may get a use of un-initialized if `keyA` fails to init on cleanup.